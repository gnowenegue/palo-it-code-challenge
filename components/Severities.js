import React, { useContext } from 'react';
import Link from 'next/link';

import AppContext from '../contexts/AppContext';

import Severity from './Severity';

const Severities = () => {
  const { setSeverity, illness, severity } = useContext(AppContext);

  const emoticons = ['ðŸ¤¤', 'ðŸ¥´', 'ðŸ¤¢', 'ðŸ¤®', 'ðŸ‘¿'];

  return (
    <div id="severity" className="modal">
      <div className="modal-content">
        <h5 className="center-align">
          {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
          How severe is your <i className="red-text">{illness.name}</i>?
        </h5>
        <div className="row">
          {emoticons.map((e, i) => {
            return (
              <Severity
                // eslint-disable-next-line react/no-array-index-key
                key={i}
                id={i + 1}
                emoticon={e}
                setSeverity={setSeverity}
              />
            );
          })}
        </div>
      </div>
      <div className="modal-footer center-align">
        <Link
          href={`/hospitals?severity=${severity}`}
          // as={`/hospitals/${severity}`}
        >
          <a
            href="#!"
            className="waves-effect waves-red modal-close btn disabled"
          >
            Which hospitals?
          </a>
        </Link>
      </div>
      <style jsx>
        {`
          .modal .modal-content h5 {
            margin-bottom: 24px;
          }
          .modal .modal-content .row {
            margin-bottom: 0;
          }
          .modal .modal-footer {
            padding-bottom: 24px;
            height: auto;
          }
          .modal .modal-footer.center-align {
            text-align: center;
          }
        `}
      </style>
    </div>
  );
};

export default Severities;
