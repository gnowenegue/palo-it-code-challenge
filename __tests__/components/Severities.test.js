import React from 'react';
import renderer from 'react-test-renderer';
// import { mount, shallow, render } from 'enzyme';

import AppContext from '../../contexts/AppContext';
import Severities from '../../components/Severities';

describe('<Severities />', () => {
  test('Snapshot test', () => {
    const render = renderer.create(
      <AppContext.Provider
        value={{
          setSeverity: () => {},
          illness: { id: 1, name: 'flu' },
        }}
      >
        <Severities />
      </AppContext.Provider>
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
