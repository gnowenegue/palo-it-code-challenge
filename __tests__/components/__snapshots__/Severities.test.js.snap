// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Severities /> Snapshot test 1`] = `
<div
  className="jsx-4257545236 modal"
  id="severity"
>
  <div
    className="jsx-4257545236 modal-content"
  >
    <h5
      className="jsx-4257545236 center-align"
    >
      How severe is your 
      <i
        className="jsx-4257545236 red-text"
      >
        flu
      </i>
      ?
    </h5>
    <div
      className="jsx-4257545236 row"
    >
      <div
        className="jsx-3703842634 col s2 center-align offset-s1"
      >
        <input
          className="jsx-3703842634"
          id="severity-1"
          name="severity"
          type="radio"
          value={1}
        />
        <label
          className="jsx-3703842634 blue-text text-darken-4"
          htmlFor="severity-1"
        >
          ðŸ¤¤
          <br
            className="jsx-3703842634"
          />
          1
        </label>
      </div>
      <div
        className="jsx-3703842634 col s2 center-align "
      >
        <input
          className="jsx-3703842634"
          id="severity-2"
          name="severity"
          type="radio"
          value={2}
        />
        <label
          className="jsx-3703842634 blue-text text-darken-4"
          htmlFor="severity-2"
        >
          ðŸ¥´
          <br
            className="jsx-3703842634"
          />
          2
        </label>
      </div>
      <div
        className="jsx-3703842634 col s2 center-align "
      >
        <input
          className="jsx-3703842634"
          id="severity-3"
          name="severity"
          type="radio"
          value={3}
        />
        <label
          className="jsx-3703842634 blue-text text-darken-4"
          htmlFor="severity-3"
        >
          ðŸ¤¢
          <br
            className="jsx-3703842634"
          />
          3
        </label>
      </div>
      <div
        className="jsx-3703842634 col s2 center-align "
      >
        <input
          className="jsx-3703842634"
          id="severity-4"
          name="severity"
          type="radio"
          value={4}
        />
        <label
          className="jsx-3703842634 blue-text text-darken-4"
          htmlFor="severity-4"
        >
          ðŸ¤®
          <br
            className="jsx-3703842634"
          />
          4
        </label>
      </div>
      <div
        className="jsx-3703842634 col s2 center-align "
      >
        <input
          className="jsx-3703842634"
          id="severity-5"
          name="severity"
          type="radio"
          value={5}
        />
        <label
          className="jsx-3703842634 blue-text text-darken-4"
          htmlFor="severity-5"
        >
          ðŸ‘¿
          <br
            className="jsx-3703842634"
          />
          5
        </label>
      </div>
    </div>
  </div>
  <div
    className="jsx-4257545236 modal-footer center-align"
  >
    <a
      className="jsx-4257545236 waves-effect waves-red modal-close btn disabled"
      href="#!"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Which hospitals?
    </a>
  </div>
</div>
`;
